### ISSUE TITLE
Are we copying modules to build folder each time?

### ISSUE THREAD

**Issue by vivekb91** [CodeCombiner/native-android#51]
It looks like we are always deleting build folders and copying modules to build folder. If we can use links then edit with android studio project will reflect in original modules, this will be helpful for debugging.

If we are always deleting and creating the project, I doubt will this effect this build time ?

**Comment by CodeCombiner** [comment]
Copying takes about 30 seconds (on SSD), depending on disk characteristics. Average least build time is 4 minutes 30 seconds, depending on CPU. The main time is consumed by Java sources compilation. Linking files might save 1/9 of total build time but as you say it shell be helpful for debugging.

Linking will require to review building steps and to provide some changes. Also it will be useful to review and discuss current devkit building DAG. Which we have now after mixed in gradle required changes. It is helpful to make sure the tasks have correct order and are not launched in parallel when it is restricted.

**Comment by rampr** [comment]
Alex, he wanted to understand if we don't copy, but use symbolic links, then if the project is opened in Android Studio and when changes are made they will reflect in the main repo.
That way changes done in Android Studio can be brought back to the repo immediately

**Comment by CodeCombiner** [comment]
This is correct point of view and this will save development time. It will require some changes and building steps review to implement this.

**Comment by rampr** [comment]
We can do this after it is merged, I will create an issue in github.

**Comment by rampr** [comment]
Didn't realize, **this** is the issue in github :)
We can work on it after the current changes are merged

**Comment by vivekb91** [comment]
Okey. Fine

**Comment by rampr** [comment]
This looks like a duplicate of #65. Please confirm. If that is the case, we just need only 1 issue

**Comment by CodeCombiner** [comment]
These copied dirs are different but both require use of syslinks

**Comment by CodeCombiner** [comment]
New PRs for this issue
https://github.com/hashcube/native-android/pull/48
https://github.com/hashcube/devkit-core/pull/67 (Updated with new commits)

Short description:
It is still required to copy AndroidSeed project and then modify it. But it is not copied everytime when `devkit debug native-android` is called anymore as it is required in issue description. The JavaScript contents is rewrited everytime only.

Next small improvement can be added is to check if the files should be rewritten with streams in devkit-core :
```
api.streams.create('spriter');
api.streams.create('sound-map');
api.streams.create('app-js', { ...
```

@rampr , @vivekb91 let me know, please consider more improvements and optimizations. I will continue after your feedbacks.

**Comment by rampr** [comment]
@CodeCombiner Can modules(like freshchat etc..) also be symlinked ?

**Comment by CodeCombiner** [comment]
Yes, I think we can symlink all jars and aars.
